common/src/main/java/org/keycloak/common/util/StringSerialization.java

```diff
@@ -0,0 +1,116 @@
+package org.keycloak.common.util;
+
+import java.net.URI;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Utilities to serialize objects to string. Type safety is not guaranteed here and is responsibility of the caller.
+ * @author hmlnarik
+ */
+public class StringSerialization {
```

Comment:
If I understood it correctly,  you wrote this to create a string representation from an object state (or any set of input values) so that you can later parse it later and reconstruct the object state.

Out of curiosity, wouldn't be easier to just parse a pre-defined string format using some well defined separator ?

Reply:
Exactly. This task occurs from time to time (now e.g. in #6127) so it is beneficial to have pre-defined utility to cover it.

Reply:
Understood